### ISSUE TITLE
:bug: Minor issue and a question

### ISSUE THREAD

**Issue by adir1** [nunocoracao/blowfish#391]
First - Awesome job, really well done!!!
I went through Many tools and templates, starting with Gatsby echo, via NextJS, and ended up here with Blowfish!

Now issue: I took very long time to figure out why Recent was showing empty on home page. Turns out all my posts had type of "post" (from the WP to MD export tool I used). So that explains why changing directory to "blog" and playing with mainSections param didn't help at first. Eventually I set mainSections to "post" and that did it.
Btw - all that talk about "section with most pages" doesn't seem to apply - I definitely have most pages in posts dir/section.

Finally a Question- I want to add Photo Gallery to my blog, something where I can simply drop new photo images as I take em, not related to any specific posts. Any suggestions how to do this?

**Comment by nunocoracao** [comment]
Hey @adir1 thanks

issue - not sure I understand the problem. Will need some time to investigate. If you can provide some examples of the problem it will make my life easier. 

regarding the photo gallery, blowfish doesn't offer that (atm at least). You can either extend the theme using your own html/js code, import a JS library and code the respective html, or get an additional hugo theme in your site (in theory you can import more than one theme). AFAIK some theme projects just provide shortcuts/partials for very specific use-cases. 

e.g. https://github.com/Darthagnon is working on https://github.com/Darthagnon/hugo-easy-gallery which might be a solution

**Comment by adir1** [comment]
Thanks a lot - I'll look further into gallery.
Regarding the issue - all my posts (converted via Hugo WP exit tool) have type of "post" in the markdown files:
type: post

Directory they are all sitting in is "posts", but it doesn't matter since type is "post" explicitly in all MD files.
So solution for displaying the recent posts correctly for me was to add this in params.toml
mainSections = [ "post","posts" ]

I agree it may not really be an issue, just as a newbie to Hugo took me some time to track down why Recents was empty while all Posts are displayed fine via Posts link and Tags, etc. The default for mainSections is "posts" it seems, so query was excluding all entries with type="post".

**Comment by nunocoracao** [comment]
Should work just with posts if you restart Hugo server no?

**Comment by nunocoracao** [comment]
@adir1 do you have a repo I can clone and test locally?

**Comment by adir1** [comment]
Thanks a lot! Still totally work in progress, but I pushed the latest snapshot now: https://github.com/adir1/adir1hugo

**Comment by adir1** [comment]
While converting some more pages today, I tried TypeIt shortcode - doesn't seem to work sadly. Browser console says: "Uncaught ReferenceError: TypeIt is not defined"
I am pointing to latest version, I did "hugo mod get -u" earlier today.

**Comment by adir1** [comment]
More fun times - I am new to Go Lang and Hugo, so took some time...
Tracked down another small bug where **heroStyle** param doesn't work at a single Post level,
I can do a PR later against Dev, or just replace line 6 in single.html with:
  {{ $heroStyle := .Site.Params.article.heroStyle }}
  {{ if .Params.heroStyle }}
  {{ $heroStyle = .Params.heroStyle }}
  {{ end }}
  {{ $heroStyle := print "partials/hero/" $heroStyle ".html" }}

**Comment by nunocoracao** [comment]
Hello @adir1 happy to discuss all of the points here with you or send you the right links for docs for some of them. 

However, simply saying several things don't work, inside the same ticket, without providing steps to reproduce or a repo for me to validate your remarks is not helping us solve any issues you are facing. Even if they are not bugs.

I'll take a look at this thread later today. If you can please provide details for each single point (or a repo).

**Comment by adir1** [comment]
I agree, no worries - I don't like to spam issues also, will just do PR for some, I think I am getting the hang of things more. Here's the repo again: https://github.com/adir1/adir1hugo

**Comment by nunocoracao** [comment]
Hey @adir1 cloned your repo, here are my comments on each topic you raised

On the Post/Posts issue, simple workaround is there, but see in repo above as time allows
- Blowfish doesn't support the type front matter parameter. If you removed that and keep the "posts" folder name inside content everything works as expected. I tested in in your repo and it works.

Issue with heroStyle, I am going to do a PR tonight
- Opened https://github.com/nunocoracao/blowfish/issues/405

TypeIt not working - not critical, but take a look as time allows. It generates HTML from shortcode, but include for TypeIt JS library is not there for some reason. Don't know if I am supposed to activate something somewhere. 
- Opened https://github.com/nunocoracao/blowfish/issues/406 

Closing this one and will use the two new ones to track developments.
